src/Datasource/Query.php

```diff
@@ -396,7 +666,7 @@ public function set($fields = null)
     /**
      * @inheritDoc
      */
-    public function offset($offset)
+    public function offset(?int $offset): Query|QueryInterface
```

Comment:
Remove return types from all functions that return `$this`.

Reply:
Again, this is not good practice. I've changed the return type to what the interface requires. Our implementation should be in line with what the interface wants. Otherwise, there is not much use in implementing it.

Reply:
Cake's convention is to skip return type for all methods that return `$this` so please just follow that.